src/main/java/bgu/spl/mics/application/passiveObjects/Diary.java

```diff
@@ -11,24 +24,36 @@
  * You can add ONLY private fields and methods to this class as you see fit.
  */
 public class Diary {
+	private static Diary _instance = new Diary();
+	private List<Report> _diary;
+	private ReadWriteLock _lock;
+	private AtomicInteger _total;
+
 	/**
 	 * Retrieves the single instance of this class.
 	 */
 	public static Diary getInstance() {
-		//TODO: Implement this
-		return null;
+		return _instance;
+	}
+
+	private Diary() {
+		_diary = new ArrayList<>();
+		_lock = new ReentrantReadWriteLock();
```

Comment:
please explain what's that and why do you use it
a. to me :)
b. as a comment in the code

Reply:
I am adding a comment. It is also explained [here](https://www.cs.bgu.ac.il/~spl201/index.php?page=Thread-safty2#readerswriters_example)